### DISCUSSION TITLE
JKBMS reports wrong current and soc. Can serialbattery use Multiplus reported/calculated current/soc?

### DISCUSSION THREAD

**Discussion by thegitarist** [Louisvdw/dbus-serialbattery#496] [Q&A] [answered]
Hi everyone,

my JKBMS reports wrong current in some load situations and this results in wrong soc and (dis)charge parameters.
![image](https://user-images.githubusercontent.com/1675245/225381690-7e91e1ad-4f6c-46ab-a8bc-da0e2b9e8ff0.png)

Is it possible to use the Multiplus reported/calculated current and soc?
![image](https://user-images.githubusercontent.com/1675245/225381453-1f254294-9c17-469a-95ed-a4bbf5b600c7.png)

Maybe someone else is experiencing this strange behavior.

BR
luke

**Comment by mr-manuel** [comment, marked as answer]
Hi,

there is a feature request opened for combining both: https://github.com/Louisvdw/dbus-serialbattery/issues/403

But you are asking, if you can use the MultiPlus as battery monitor instead of the BMS. Am I correct?

**Reply by thegitarist** [reply]
#403 is exactly my use case

**Reply by mr-manuel** [reply]
Please don't forget to mark this as the correct answer.

**Comment by LarsAndersenDenmark** [comment]
I would rather have the option of the system using the SOC from the shunt and still being able to use parameters forwarded by the BMS, i.e. controlling charge current, setting CVL, DVL based on the BMS individual cell values. cells are of course balanced as long as they are new, but over time, the cell will deviate which could be controlled and shown by the system.

**Reply by mr-manuel** [reply]
This is probably a very specific use case. You can achieve this by using a virtual battery and feed the data over MQTT (from Node-RED and Venus-OS Large for example). It allows you to combine the data you want.

You can find the driver here [venus-os_dbus-mqtt-battery](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery)

**Reply by thegitarist** [reply]
this is exactly what I'm doing as a workaround. I am using the cell voltage to control charge and discharge current with serialbattery. The current and voltage of my multiplus is good enough, at least for my usecase, so I want avoid using the shunt.

**Reply by LarsAndersenDenmark** [reply]
Hi thegitarist

What is the workaround you have done then?
My aim is to control charge current at nearly fully battery better than victron when running ESS. First was based on SOC regulating grid setpoint, but SOC is just extremely off and is often only 70-80% when battery voltage is at CVL and when cell voltage is at max (OVP by JKBMS) and report from overvoltage is given from the BMS. Controlling on CVL is a bit better, but the code does not work properly when my infeed is high.
Often at infeed, my JK BMS do not register amps below 20A eventhough I can observe cell voltages increase.

**Reply by LarsAndersenDenmark** [reply]
> This is probably a very specific use case. You can achieve this by using a virtual battery and feed the data over MQTT (from Node-RED and Venus-OS Large for example). It allows you to combine the data you want.
> 
> 
> 
> You can find the driver here [venus-os_dbus-mqtt-battery](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery)

Well, actually just need a grid connected solar system with feedin & battery and liked the VRM portal.

Victron is expensive, so expected it was a good system...also for ESS. The system do though charge agressively near full battery since I have 2x mppts 250/100 and only 1 mp2 48/5000 since that fits house power demand. Feedin is not the goal, but is of course a way to pay off the investment:-)

**Reply by thegitarist** [reply]
> Hi thegitarist
> 
> What is the workaround you have done then? My aim is to control charge current at nearly fully battery better than victron when running ESS. First was based on SOC regulating grid setpoint, but SOC is just extremely off and is often only 70-80% when battery voltage is at CVL and when cell voltage is at max (OVP by JKBMS) and report from overvoltage is given from the BMS. Controlling on CVL is a bit better, but the code does not work properly when my infeed is high. Often at infeed, my JK BMS do not register amps below 20A eventhough I can observe cell voltages increase.

I have enabled CCCM_CV_ENABLE and disabled CCCM_SOC_ENABLE. I was experiencing the same issues as u describe with my JKBMS. In the graph u can see the difference between Victron Shunt and JKBMS reported current. If the current is below ~ -40A, JKBMS reports 0A. The result was extreme off SOC @ CVL.

This strange situation mainly occurs, when I have high ac output on multiplus and high dc input from mppt

**Reply by LarsAndersenDenmark** [reply]
> > Hi thegitarist
> > What is the workaround you have done then? My aim is to control charge current at nearly fully battery better than victron when running ESS. First was based on SOC regulating grid setpoint, but SOC is just extremely off and is often only 70-80% when battery voltage is at CVL and when cell voltage is at max (OVP by JKBMS) and report from overvoltage is given from the BMS. Controlling on CVL is a bit better, but the code does not work properly when my infeed is high. Often at infeed, my JK BMS do not register amps below 20A eventhough I can observe cell voltages increase.
> 
> I have enabled CCCM_CV_ENABLE and disabled CCCM_SOC_ENABLE. I was experiencing the same issues as u describe with my JKBMS. In the graph u can see the difference between Victron Shunt and JKBMS reported current. If the current is below ~ -40A, JKBMS reports 0A. The result was extreme off SOC @ CVL.
> 
> This strange situation mainly occurs, when I have high ac output on multiplus and high dc input from mppt

Exactly my experience also on high inputs! But have you chosen the shunt as the active batterymonitor ? and how do you then control the voltage of the cells from the JKBMS? By NodeRed? I plan to try using the shunt as active monitor and hope it can do SOC well. Then I have a NodeRed script handling max charge to battery depending on the SOC level. But I haven't figured out how to control if any cell voltages get out of spec. Have you??

The shunt can only see the total voltage. I also have a NodeRed based on the input for CCL based on the JKBMS voltages and Louisvdw driver (Only cell voltage enable and SOC disabled), but for now it seems not stable enough and sometimes my battery do not charge...maybe due to JKBMS SOC is wrong and JKBMS current is also wrong....and at some point disturbing the Victron system.

**Reply by thegitarist** [reply]
For current control I am using the serialbattery driver. There is a section where u can set current depending on cell voltage:

```
# Charge current control management referring to cell-voltage enable (True/False).
CCCM_CV_ENABLE = True
# Discharge current control management referring to cell-voltage enable (True/False).
DCCM_CV_ENABLE = True

# Set Steps to reduce battery current. The current will be changed linear between those steps
CELL_VOLTAGES_WHILE_CHARGING         = [3.65, 3.6, 3.51, 3.5, 3.4]
MAX_CHARGE_CURRENT_CV                = [   0,   1,   2,  50, 100]

CELL_VOLTAGES_WHILE_DISCHARGING      = [2.70, 2.9, 3.0, 3.1]
MAX_DISCHARGE_CURRENT_CV             = [   0,   5,  30, 100]
```

I am still playing around with these values but at the end, my strategy here is to drop max current to <2A at ~ max cell volt, so the BMS can compensate this.

This approach works for me, but the SOC is still sometimes kind of off, specially when the BMS kicks in and disconnects the battery (voltage & SOC  rise to MP limit set in veconfig). Sure I can use Victron Shunt, but the best part is no part, and I wanted to reduce complexity.

I am using nodeRed just as mqtt > prometheus bridge to collect all metrics. Sure u can control everything with nodeRed as well, but my setup works so far.
